pkg/vf/rosetta_arm64_test.go

```diff
@@ -0,0 +1,111 @@
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityNotSupported
+		},
+		errorValue: "rosetta is not supported",
+	},
+	"TestRosettaInstalled": {
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityInstalled
+		},
+	},
+	"TestRosettaNotInstalled-NotToBeInstalled": {
+		installRosetta: false,
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityNotInstalled
+		},
+		errorValue: "rosetta is not installed",
+	},
+	"TestRosettaNotInstalled-InstallationCancelledButIgnoreIfMissingFalse": {
+		installRosetta:  true,
+		ignoreIfMissing: false,
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityNotInstalled
+		},
+		doInstallRosetta: func() error {
+			return fmt.Errorf("VZErrorDomain Code=%d", vz.ErrorOperationCancelled)
+		},
+		errorValue: fmt.Sprintf("failed to install rosetta: VZErrorDomain Code=%d", vz.ErrorOperationCancelled),
+	},
+	"TestRosettaNotInstalled-InstallationCancelledButIgnoreIfMissingTrue": {
+		installRosetta:  true,
+		ignoreIfMissing: true,
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityNotInstalled
+		},
+		doInstallRosetta: func() error {
+			return fmt.Errorf("VZErrorDomain Code=%d", vz.ErrorOperationCancelled)
+		},
+	},
+	"TestRosettaNotInstalled-InstallationFailed": {
+		installRosetta:  true,
+		ignoreIfMissing: true,
+		checkRosettaDirectoryShareAvailability: func() vz.LinuxRosettaAvailability {
+			return vz.LinuxRosettaAvailabilityNotInstalled
+		},
+		doInstallRosetta: func() error {
+			return fmt.Errorf("VZErrorDomain Code=%d", vz.ErrorInstallationFailed)
+		},
+		errorValue: fmt.Sprintf("failed to install rosetta: VZErrorDomain Code=%d", vz.ErrorInstallationFailed),
+	},
+}
+
+func TestCheckRosettaAvailability(t *testing.T) {
+	t.Run("name", func(t *testing.T) {
+		for name := range checkRosettaAvailabilityTests {
+			t.Run(name, func(t *testing.T) {
+				test := checkRosettaAvailabilityTests[name]
+				testCheckRosettaAvailability(t, &test)
+			})
+		}
+	})
+}
+
+func testCheckRosettaAvailability(t *testing.T, test *checkRosettaAvailabilityTest) {
+	rosetta :=
+		RosettaShare{
+			InstallRosetta:  test.installRosetta,
+			IgnoreIfMissing: test.ignoreIfMissing,
+			DirectorySharingConfig: config.DirectorySharingConfig{
+				MountTag: "mount",
+			},
+		}
+
+	origCheckRosettaDirectoryShareAvailability := checkRosettaDirectoryShareAvailability
+	checkRosettaDirectoryShareAvailability = test.checkRosettaDirectoryShareAvailability
+	origDoInstallRosetta := doInstallRosetta
+	doInstallRosetta = test.doInstallRosetta
+	defer func() {
+		checkRosettaDirectoryShareAvailability = origCheckRosettaDirectoryShareAvailability
+		doInstallRosetta = origDoInstallRosetta
+	}()
```

Comment:
Not sure you need to restore the initial values after running the test? I'd expect each `xxx_test.go` file to run in its own process (to be checked though ;)

Reply:
So, I've been digging on it a bit and I didn't find any clear response but by looking at the code it seems that each test is run by launching a go routine https://github.com/golang/go/blob/182bdbb1e1d62fe53f8386e7b02c7738898d373b/src/testing/testing.go#L555 and from what I read all goroutines within a Go program share the same process so all tests run in the same process.
By creating a new test file I can see it to be true so I think it's better to keep the defer to avoid problems for future tests.

Reply:
`go help test` has some details on this:
> Each listed package causes the execution of a separate test binary.
so all tests in the same dir will run in parallel in go routine.
Tests in different dirs should run in separate processes.

Reply:
No need to make any changes though :)